gluefactory/datasets/oxford_paris_mini_1view_jpldd.py

```diff
@@ -253,15 +296,110 @@ def __getitem__(self, idx):
+        if self.num_select_with_current_scale % self.relevant_batch_size == 0:
+            self.num_select_with_current_scale = 0  # if batch changes set counter to 0
+            return True
+        else:
+            return False
+        
+
+    def select_resize_shape(self, original_img_size: tuple):
+        """
+        Depending on whether resize or multiscale learning is activated the shape to resize the
+        image to is returned. If none of it is activated, the original image size will be returned.
+        Reshape has prevalence over multiscale learning!
+        """
+        do_reshape = self.conf.reshape is not None
+        do_ms_learning = self.conf.multiscale_learning.do
+        if not do_reshape and not do_ms_learning:
+            return original_img_size
+
+        if do_reshape:
+            return int(self.conf.reshape)
+
+        if do_ms_learning:
+            if self.do_change_size_now():
+                self.num_select_with_current_scale += 1
+                scales_list = self.conf.multiscale_learning.scales_list
+                scale_selection = self.conf.multiscale_learning.scale_selection
+                assert len(scales_list) > 1 # need more than one scale for multiscale learning to make sense
+
+                if scale_selection == "random":
+                    choice = int(random.choice(scales_list))
+                    self.current_scale = choice
+                    return choice
+                elif scale_selection == "round-robin":
+                    current_scale = scales_list[self.scale_selection_idx]
+                    self.current_scale = current_scale
+                    self.scale_selection_idx += 1
+                    self.scale_selection_idx = self.scale_selection_idx % len(scales_list)
+                    return int(current_scale)
+            else:
+                self.num_select_with_current_scale += 1
+                return self.current_scale
+
+        raise Exception("Shouldn't end up here!")
+    
+    
+    def set_num_selected_with_current_scale(self, value: int) -> None:
+        """
+        Sets the self.num_selected_with_current_scale variable to a certain value.
+        This method is implemented as interface for the MergedDataset to be able to deal with multiscale learning
+        on multiple datasets
+
+        Args:
+            value (int): new value for variable
+        """
+        self.num_select_with_current_scale = value
+
+
+    def get_current_scale(self) -> int:
+        """
+        Returns the current used scale to reshape images to. Returns None if multiscale learning is deactivated.
+        This method is implemented as interface for the MergedDataset to be able to deal with multiscale learning
+        on multiple datasets
+
+        Returns:
+            int: current scale used to reshape in multi-scale training. None if its deactivated
+        """
+        return self.current_scale
+    
+    
+    def set_current_scale(self, value):
```

Comment:
Couldn't we put this method and the few above in a parent class and reuse these methods that way? I feel like there is a lot of code duplication for each dataset

Reply:
I think that this could be an idea. but I would do it after paper submission as the time untiul submission can be used better. I will open an Issue for later once this is merged.

Reply:
Agreed